pkg/fileutils/fileutils.go

```diff
@@ -1,26 +1,53 @@
 package fileutils
 
 import (
-	log "github.com/Sirupsen/logrus"
 	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
 )
 
 // Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
 func Matches(relFilePath string, patterns []string) (bool, error) {
-	for _, exclude := range patterns {
-		matched, err := filepath.Match(exclude, relFilePath)
+
+	matched := false
+
+	for _, pattern := range patterns {
+
+		var negative bool
+		if pattern == "" {
+			continue
+		}
+
+		if pattern[0] == '!' {
+			negative = true
+			pattern = pattern[1:]
+		}
```

Comment:
`!` would mean nothing. As in don't match anything. Should I keep it or is it confusing?